oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/InMemoryReactiveOAuth2AuthorizedClientService.java

```diff
@@ -62,8 +62,19 @@ public <T extends OAuth2AuthorizedClient> Mono<T> loadAuthorizedClient(String cl
 		Assert.hasText(clientRegistrationId, "clientRegistrationId cannot be empty");
 		Assert.hasText(principalName, "principalName cannot be empty");
 		return (Mono<T>) this.clientRegistrationRepository.findByRegistrationId(clientRegistrationId)
-			.map((clientRegistration) -> new OAuth2AuthorizedClientId(clientRegistrationId, principalName))
-			.flatMap((identifier) -> Mono.justOrEmpty(this.authorizedClients.get(identifier)));
+				.mapNotNull((clientRegistration) -> {
+					OAuth2AuthorizedClientId id = new OAuth2AuthorizedClientId(clientRegistrationId, principalName);
+					OAuth2AuthorizedClient cachedAuthorizedClient = this.authorizedClients.get(id);
+					if (cachedAuthorizedClient == null) {
+						return null;
+					}
+					// @formatter:off
+					return new OAuth2AuthorizedClient(clientRegistration,
+							cachedAuthorizedClient.getPrincipalName(),
+							cachedAuthorizedClient.getAccessToken(),
+							cachedAuthorizedClient.getRefreshToken());
+					// @formatter:on
+				});
```

Comment:
@sjohnr suggested in https://github.com/spring-projects/spring-security/issues/15511#issuecomment-2484441982 that `flatMap` was preferred here, but I disagree, because a single `mapNotNull` surely is less expensive.
However, if code style/consistency is more important here, I'm happy to refactor this.

Reply:
Okay then I'll leave it as it is. Feel free to polish afterwards.